### PR TITLE
Modify avatar image fallback component

### PR DESCRIPTION
**Pull request by SKona306**
This will modify the avatar image fallback component by:

- changing name from `avatar-image-fallback` to `app-avatar-image-fallback` 
- modify logic to use ngLetterAvatar any time an `img` element throws an error
- modify logic to use default user avatar when no `name` or `src` is provided 

image

### CODE REVIEW

File: app/components/avatar-image-fallback/avatar-image-fallback.component.ts [hunk 1]
```diff
@@ -22,31 +22,39 @@ class AvatarImageFallback implements AvatarImageFallbackBindings {
...
-        });
-    }
-
-    setCharCount() {
-        if (this.name) {
-            this.charCount = this.name.includes(' ') ? 2 : 1;
+        } else {
+            this.$element.find('img').bind('error', () => {
+                this.$element[0].children[0].remove();
+            });
```
Comment (Morkos) [inline app/components/avatar-image-fallback/avatar-image-fallback.component.ts]: so the broken image gets removed? However, does the letter avatar still show when an image fails to load?

File: app/components/avatar-image-fallback/avatar-image-fallback.component.ts [hunk 2]
```diff
@@ -22,31 +22,39 @@ class AvatarImageFallback implements AvatarImageFallbackBindings {
...
         }
     }
 }
 
-angular.module('otr-ui-shared-components').component('avatarImageFallback', {
+angular.module('otr-ui-shared-components').component('appAvatarImageFallback', {
```
Comment (Morkos) [inline app/components/avatar-image-fallback/avatar-image-fallback.component.ts]: glad you caught this! e.g. adding "app" as a prefix

File: app/components/avatar-image-fallback/avatar-image-fallback.component.ts [hunk 3]
```diff
@@ -22,31 +22,39 @@ class AvatarImageFallback implements AvatarImageFallbackBindings {
...
+
         this.setCharCount();
+        this.setLetterProfileAvatar();
+    }
 
-        this.$element.find('img').bind('error', () => {
+    //sets ngLetterAvatar character count; max 2 characters
+    setCharCount() {
+        if (this.name) {
+            this.charCount = this.name.includes(' ') ? 2 : 1;
```
Comment (Morkos) [inline app/components/avatar-image-fallback/avatar-image-fallback.component.ts]: use a regex here for "space characters" (includes single/multiple tabs and spaces) to make it more robust.

`/s+/.test(this.name) ? 2 : 1.
`
See here for example:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test

**Review by Morkos — CHANGES_REQUESTED** [review]
Better! See comments below

File: app/components/avatar-image-fallback/avatar-image-fallback.component.ts (hunk 1 above)
Comment (SKona306) [reply app/components/avatar-image-fallback/avatar-image-fallback.component.ts]: I added a new variable that will show the letter avatar when the image fails to load

**Comment by SKona306** [comment]
I changed the name of the class to avoid confusion on parent applications

File: app/components/avatar-image-fallback/app-avatar-image-fallback.component.ts [hunk 4]
```diff
@@ -29,8 +30,9 @@ class AvatarImageFallback implements AvatarImageFallbackBindings {
 
     //sets ngLetterAvatar character count; max 2 characters
     setCharCount() {
+        const regex = /\s+/;
```
Comment (Morkos) [inline app/components/avatar-image-fallback/app-avatar-image-fallback.component.ts]: const spacePattern = /\s+/;

File: index-test.html [hunk 5]
```diff
@@ -108,7 +108,7 @@
                     $rootScope.profileImageUrl = "https://graph.facebook.com/v2.8/3668555106576865/picture?height=961"
                     $rootScope.name = "James Smith"
                     $rootScope.avatarClass = "customer"
-                    $rootScope.defaultUserProfileImage = "https://s3.amazonaws.com/otr-assets/img/icons/default-user.png"
+                    $rootScope.defaultUserProfileImage = "https://s3.amazonaws.com/otr-assets/img/icons/jkfkl;jdaf"
```
Comment (Morkos) [inline index-test.html]: oof! lol

File: app/components/avatar-image-fallback/app-avatar-image-fallback.component.ts [hunk 6]
```diff
@@ -15,8 +15,9 @@ class AvatarImageFallback implements AvatarImageFallbackBindings {
     // interface
     imageClass!: string;
     charCount!: number;
+    isLetterAvatarShowing!: boolean;
```
Comment (Morkos) [inline app/components/avatar-image-fallback/app-avatar-image-fallback.component.ts]: rename to isLetterAvatarVisible

**Review by Morkos — APPROVED** [review]
Looking good. Conditional approval -- fix comments below before doing squash & merge into master.

**Pull request merged**